app/controllers/openid_application_controller.rb

```diff
@@ -106,7 +115,10 @@ def establish_session(profile)
     ttl = token.payload['exp'] - Time.current.utc.to_i
     user_identity = OpenidUserIdentity.build_from_okta_profile(uuid: token.identifiers.uuid, profile: profile, ttl: ttl)
     @current_user = OpenidUser.build_from_identity(identity: user_identity, ttl: ttl)
-    @session = build_session(ttl)
+    @session = build_session(ttl,
+                             Okta::UserProfile.new({ 'last_login_type' => profile['last_login_type'],
+                                                     'SecID' => profile['SecID'], 'VistaId' => profile['VistaId'],
+                                                     'npi' => profile['npi'], 'icn' => profile['icn'] }))
     @session.save && user_identity.save && @current_user.save
```

Comment:
One thing to keep in mind, currently the `derived_loa` method in user_profile.rb doesn't handled SSOi.  Which results in the user_identity object failing validation and not saving to the cache.  And with that failure the `@current_user` save isn't attempted either.

For token validation by itself, not really a problem.  But something to keep in mind in case in the future an SSOi user wanted to use Veteran Verification API.  In that case, the `@current_user` build up is required to have the necessary id's to call underlying service endpoints.